### PR TITLE
Add player.GetBySteamID and player.GetBySteamID64

### PR DESCRIPTION
**Pull request by zerfgog**
We already have player.GetByUniqueID; having these functions is nice.
Unlike #846, the implementation of these functions is not subjective.

### CODE REVIEW

File: garrysmod/lua/includes/extensions/player.lua [hunk 1]
```diff
@@ -52,7 +52,43 @@ function player.GetByUniqueID( ID )
...
+
+	return false
+
+end
+
+function player.GetBySteamID( ID )
+
+	ID = string.upper( ID )
+
+	assert( string.find( ID, "STEAM_" ), "player.GetBySteamID: passed string '" .. ID .. "' is not a valid steamid!" )
```
Comment (TheFreeman193) [inline garrysmod/lua/includes/extensions/player.lua]: I recommend you use ID:StartWith("STEAM_") here; it's more relevant to SteamIDs, and it's also considerably quicker.

**Comment by zerfgog** [comment]
ID:StartWith("STEAM_") or string.find(ID, "^STEAM_")?

**Comment by TheFreeman193** [comment]
The former; I see no need for patterns/regex here since we're only looking for absolute text values.

**Comment by wiox** [comment]
If GetByUniqueID doesn't emit errors I believe GetBySteamID and GetBySteamID64 shouldn't either.

**Comment by zerfgog** [comment]
Squashed commits.

**Pull request merged**